qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/QuteCodeActionForUnknownProperty.java

```diff
@@ -153,4 +161,46 @@ private static void doCodeActionToCreateProperty(Part propertyPart, ResolvedJava
 
 	}
 
+	private void doCodeActionsForSimilarValues(PropertyPart part, Template template, Diagnostic diagnostic,
+			ResolvedJavaTypeInfo baseResolvedType, List<CodeAction> codeActions) throws BadLocationException {
+		Collection<String> availableValues = collectAvailableValuesForPropertyPart(part, template, baseResolvedType);
+		doCodeActionsForSimilarValues(part, availableValues, template, diagnostic, codeActions);
+	}
+
+	private Collection<String> collectAvailableValuesForPropertyPart(PropertyPart node, Template template,
+			ResolvedJavaTypeInfo baseResolvedType) {
+		Collection<String> availableValues = new HashSet<>();
+		String projectUri = template.getProjectUri();
+
+		List<String> javaFieldNames = baseResolvedType.getFields().stream().map(x -> x.getName())
+				.collect(Collectors.toList());
+		for (String fieldName : javaFieldNames) {
+			availableValues.add(fieldName);
+		}
+
+		JavaTypeFilter filter = javaCache.getJavaTypeFilter(projectUri, null);
+		JavaTypeAccessibiltyRule javaTypeAccessibility = filter.getJavaTypeAccessibility(baseResolvedType,
+				template.getJavaTypesSupportedInNativeMode());
+
+		if (!isIgnoreSuperclasses(baseResolvedType, javaTypeAccessibility, filter)) {
+			List<String> extendedTypes = baseResolvedType.getExtendedTypes();
+			if (extendedTypes != null) {
+				for (String extendedType : extendedTypes) {
+					ResolvedJavaTypeInfo resolvedExtendedType = javaCache.resolveJavaType(extendedType, projectUri)
+							.getNow(null);
+					if (resolvedExtendedType != null) {
+						collectAvailableValuesForPropertyPart(node, template, resolvedExtendedType);
+					}
+				}
+			}
+		}
+
+		return availableValues;

```

Comment:
It misses methods without parameters and value resolvers. See completion for property